pallets/parachain-staking/src/lib.rs

```diff
@@ -1079,15 +1079,23 @@ pub mod pallet {
 			collator: T::AccountId,
 		) -> DispatchResultWithPostInfo {
 			let mut nominator = <NominatorState<T>>::get(&acc).ok_or(Error::<T>::NominatorDNE)?;
+			let old_total = nominator.total;
 			let remaining = nominator
 				.rm_nomination(collator.clone())
 				.ok_or(Error::<T>::NominationDNE)?;
-			ensure!(
-				remaining >= T::MinNominatorStk::get(),
-				Error::<T>::NomBondBelowMin
-			);
-			Self::nominator_leaves_collator(acc.clone(), collator)?;
-			<NominatorState<T>>::insert(&acc, nominator);
+			if nominator.nominations.0.len().is_zero() {
+				// leave the set of nominators because no nominations left
+				Self::nominator_leaves_collator(acc.clone(), collator)?;
+				<NominatorState<T>>::remove(&acc);
+				Self::deposit_event(Event::NominatorLeft(acc, old_total));
+			} else {
+				ensure!(
+					remaining >= T::MinNominatorStk::get(),
+					Error::<T>::NomBondBelowMin
+				);
+				Self::nominator_leaves_collator(acc.clone(), collator)?;
+				<NominatorState<T>>::insert(&acc, nominator);
+			}
```

Comment:
I think you should `return` early in the `if` clause. Then the stuff that is currently in the `else` clause can be moved out one level of indentation. This helps readers see that the the `if` is an edge case and we are returning to the most general flow after we check for it. Whereas leaving it all in the else clause requires a reader to keep remember the context that we're only running this code sometimes.

I don't feel strongly about it though.

Reply:
I like it better too, good recommendation